File: Frontend/HrmSystem/src/app/Models/EmployeeDTO.ts
import {EmployeeDocumentDTO} from './EmployeeDocumentDTO'
import {EmployeeEducationInfoDTO} from './EmployeeEducationInfoDTO'
import {EmployeeProfessionalCertificationDTO} from './EmployeeProfessionalCertificationDTO '
import {EmployeeFamilyInfoDTO} from './EmployeeFamilyInfoDTO '

// export interface EmployeeDTO {
//   idClient: number;
//   id: number;
//   employeeName: string;
//   employeeNameBangla?: string;
//   fatherName?: string;
//   motherName?: string;
//   idReportingManager?: number;
//   idJobType?: number;
//   idEmployeeType?: number;
//   birthDate?: Date;
//   joiningDate?: Date;
//   idGender?: number;
//   idReligion?: number;
//   idDepartment: number;
//   idSection: number;
//   idDesignation?: number;
//   hasOvertime?: boolean;
//   hasAttendenceBonus?: boolean;
//   idWeekOff?: number;
//   address?: string;
//   presentAddress?: string;
//   nationalIdentificationNumber?: string;
//   contactNo?: string;
//   idMaritalStatus?: number;
//   isActive?: boolean;
//   setDate?: Date;
//   createdBy?: string;
//   departmentName?: string;
//   designation?: string;

//   employeeDocuments: EmployeeDocumentDTO[];
//   employeeEducationInfos: EmployeeEducationInfoDTO[];
//   employeeProfessionalCertifications: EmployeeProfessionalCertificationDTO[];
//   employeeFamilyInfos: EmployeeFamilyInfoDTO[];

//   empImg?: File;
// }

export class EmployeeDTO {
  idClient: number;
  id: number;
  employeeName: string;
  employeeNameBangla?: string;
  fatherName?: string;
  motherName?: string;
  idReportingManager?: number;
  idJobType?: number;
  idEmployeeType?: number;
  birthDate?: Date;
  joiningDate?: Date;
  idGender?: number;
  idReligion?: number;
  idDepartment: number;
  idSection: number;
  idDesignation?: number;
  hasOvertime?: boolean;
  hasAttendenceBonus?: boolean;
  idWeekOff?: number;
  address?: string;
  presentAddress?: string;
  nationalIdentificationNumber?: string;
  contactNo?: string;
  idMaritalStatus?: number;
  isActive?: boolean;
  setDate?: Date;
  createdBy?: string;
  departmentName?: string;
  designation?: string;

  employeeDocuments: EmployeeDocumentDTO[];
  employeeEducationInfos: EmployeeEducationInfoDTO[];
  employeeProfessionalCertifications: EmployeeProfessionalCertificationDTO[];
  employeeFamilyInfos: EmployeeFamilyInfoDTO[];
  
  employeeImage?: string;

  constructor(data: Partial<EmployeeDTO> = {}) {
    this.idClient = data.idClient ?? 0;
    this.id = data.id ?? 0;
    this.employeeName = data.employeeName ?? '';
    this.employeeNameBangla = data.employeeNameBangla;
    this.fatherName = data.fatherName;
    this.motherName = data.motherName;
    this.idReportingManager = data.idReportingManager;
    this.idJobType = data.idJobType;
    this.idEmployeeType = data.idEmployeeType;
    this.birthDate = data.birthDate ? new Date(data.birthDate) : undefined;
    this.joiningDate = data.joiningDate ? new Date(data.joiningDate) : undefined;
    this.idGender = data.idGender;
    this.idReligion = data.idReligion;
    this.idDepartment = data.idDepartment ?? 0;
    this.idSection = data.idSection ?? 0;
    this.idDesignation = data.idDesignation;
    this.hasOvertime = data.hasOvertime ?? false;
    this.hasAttendenceBonus = data.hasAttendenceBonus ?? false;
    this.idWeekOff = data.idWeekOff;
    this.address = data.address;
    this.presentAddress = data.presentAddress;
    this.nationalIdentificationNumber = data.nationalIdentificationNumber;
    this.contactNo = data.contactNo;
    this.idMaritalStatus = data.idMaritalStatus;
    this.isActive = data.isActive ?? true;
    this.setDate = data.setDate ? new Date(data.setDate) : undefined;
    this.createdBy = data.createdBy;
    this.departmentName = data.departmentName;
    this.designation = data.designation;

    this.employeeDocuments = data.employeeDocuments ?? [];
    this.employeeEducationInfos = data.employeeEducationInfos ?? [];
    this.employeeProfessionalCertifications = data.employeeProfessionalCertifications ?? [];
    this.employeeFamilyInfos = data.employeeFamilyInfos ?? [];

    this.employeeImage = data.employeeImage;
  }

  
}

File: Backend/HanaHRM/DTO/EmployeeDTO.cs
using HanaHRM.DataAccess.Models;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace HanaHRM.DTO
{
    public class EmployeeDTO
    {
        public int IdClient { get; set; }

        public int Id { get; set; }

        public string? EmployeeName { get; set; }

        public string? EmployeeNameBangla { get; set; }

        public string? FatherName { get; set; }

        public string? MotherName { get; set; }

        public int? IdReportingManager { get; set; }

        public int? IdJobType { get; set; }

        public int? IdEmployeeType { get; set; }

        public DateTime? BirthDate { get; set; }

        public DateTime? JoiningDate { get; set; }

        public int? IdGender { get; set; }

        public int? IdReligion { get; set; }

        public int IdDepartment { get; set; }

        public int IdSection { get; set; }

        public int? IdDesignation { get; set; }

        public bool? HasOvertime { get; set; }

        public bool? HasAttendenceBonus { get; set; }

        public int? IdWeekOff { get; set; }

        public string? Address { get; set; }

        public string? PresentAddress { get; set; }

        public string? NationalIdentificationNumber { get; set; }

        public string? ContactNo { get; set; }

        public int? IdMaritalStatus { get; set; }

        public bool? IsActive { get; set; }

        public DateTime? SetDate { get; set; }

        public string? CreatedBy { get; set; }

        public string? DepartmentName { get; set; }
        public string? Designation { get; set; }
        public string? ReligionName { get; set; }
        public string? MaritalStatusName { get; set; }

        public List<EmployeeDocumentDTO>EmployeeDocuments { get; set; } = [];

        public List<EmployeeEducationInfoDTO> EmployeeEducationInfos { get; set; } = [];

        public List<EmployeeProfessionalCertificationDTO> EmployeeProfessionalCertifications { get; set; } = [];
        public List<EmployeeFamilyInfoDTO> EmployeeFamilyInfos { get; set; } = [];

       // public IFormFile? EmpImg { get; set; }
        public string? EmployeeImage { get; set; }
        public string? UploadedFileExtension { get; set; } 


    }

}
